packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js

```diff
@@ -0,0 +1,222 @@
+import SidebarNavigationItem from '../sidebar-navigation-item';
+
+const EMPTY_OBJECT = {};
+
+function TemplateAreaButton( { postId, icon, title } ) {
+	const icons = {
+		header,
+		footer,
+	};
+	const linkInfo = useLink( {
+		postType: 'wp_template_part',
+		postId,
+	} );
+
+	return (
+		<SidebarNavigationItem
+			className="edit-site-sidebar-navigation-screen-template__template-area-button"
+			{ ...linkInfo }
+			icon={ icons[ icon ] ?? layout }
+			withChevron
+		>
+			<Truncate
+				limit={ 20 }
+				ellipsizeMode="tail"
+				numberOfLines={ 1 }
+				className="edit-site-sidebar-navigation-screen-template__template-area-label-text"
+			>
+				{ decodeEntities( title ) }
+			</Truncate>
+		</SidebarNavigationItem>
+	);
+}
+
+export default function HomeTemplateDetails() {
+	const navigator = useNavigator();
+	const {
+		params: { postType, postId },
+	} = navigator;
+	const { editEntityRecord } = useDispatch( coreStore );
+
+	const {
+		allowCommentsOnNewPosts,
+		templatePartAreas,
+		postsPerPage,
+		postsPageTitle,
+		postsPageId,
+		templateRecord,
+	} = useSelect(
+		( select ) => {
+			const { getEntityRecord } = select( coreStore );
+			const siteSettings = getEntityRecord( 'root', 'site' );
+			const { getSettings } = unlock( select( editSiteStore ) );
+			const siteEditorSettings = getSettings();
+			const _templateRecord =
+				select( coreStore ).getEditedEntityRecord(
+					'postType',
+					postType,
+					postId
+				) || EMPTY_OBJECT;
+			const _postsPageRecord = siteSettings?.page_for_posts
+				? select( coreStore ).getEntityRecord(
+						'postType',
+						'page',
+						siteSettings?.page_for_posts
+				  )
+				: EMPTY_OBJECT;
+
+			return {
+				templateRecord: _templateRecord,
+				allowCommentsOnNewPosts:
+					siteSettings?.default_comment_status === 'open',
+				postsPageTitle: _postsPageRecord?.title?.rendered,
+				postsPageId: _postsPageRecord?.id,
+				postsPerPage: siteSettings?.posts_per_page,
+				templatePartAreas: siteEditorSettings?.defaultTemplatePartAreas,
+			};
+		},
+		[ postType, postId ]
+	);
+
+	const [ commentsOnNewPostsValue, setCommentsOnNewPostsValue ] =
+		useState( '' );
+	const [ postsCountValue, setPostsCountValue ] = useState( 1 );
+	const [ postsPageTitleValue, setPostsPageTitleValue ] = useState( '' );
+
+	useEffect( () => {
+		setCommentsOnNewPostsValue( allowCommentsOnNewPosts );
+		setPostsPageTitleValue( postsPageTitle );
+		setPostsCountValue( postsPerPage );
+	}, [ postsPageTitle, allowCommentsOnNewPosts, postsPerPage ] );
+
+	const templateAreas = useMemo( () => {
+		return templateRecord?.blocks && templatePartAreas
+			? templateRecord.blocks
+					.filter( ( { name } ) => name === 'core/template-part' )
+					.map( ( { attributes } ) => ( {
+						...templatePartAreas?.find(
+							( { area } ) => area === attributes?.tagName
+						),
+						...attributes,
+					} ) )
+			: [];
+	}, [ templateRecord?.blocks, templatePartAreas ] );
```

Comment:
This currently only displays template part areas that are at the root of the document, so for a site that has nested the header or footer template parts within a Group block, they don't appear to show up. In the following screenshot, the site's header is wrapper in a Group block, but the Footer isn't, so the Footer is showing but not the Header:

![image](https://github.com/WordPress/gutenberg/assets/14988353/ca62a718-6b18-415a-a04b-d158e8ff0682)

Would it be worth re-using the [`getFilteredTemplatePartBlocks` utility function](https://github.com/WordPress/gutenberg/blob/a065bb5263514f669b1662da9790557980e0f952/packages/edit-site/src/store/utils.js#L69) we use in the right hand template sidebar within the site editor, to iterate into nested template part areas?

This is a bit of a nit, and perfectly fine to address in a follow-up if you'd prefer, since this PR is already pretty big.

Reply:
> Would it be worth re-using the [getFilteredTemplatePartBlocks utility function](https://github.com/WordPress/gutenberg/blob/a065bb5263514f669b1662da9790557980e0f952/packages/edit-site/src/store/utils.js#L69) we use in the right hand template sidebar within the site editor, to iterate into nested template part areas?

If it's already in there and tested, then I think it's a low risk. 👍

Reply:
Actually, I might add it in a follow up. I wanted to add the last updated footer to template parts as well 🤔 

You're right, it's big enough 😄